### ISSUE TITLE
Is this implementation split validation dataset?

### ISSUE THREAD

**Issue by sh0416** [graytowne/caser_pytorch#7]
I think it doesn't do validation test..

Can I see which part of implement is creating validation dataset and where to validation??

**Comment by graytowne** [comment]
You can find the validation dataset at datasets/ml1m/validation. You can simply change the train_root and test_root to the validation files to work on validation. 
Our hyper-parameters (learning rate, activations, etc) are searched through the performance on validation set.

**Comment by sh0416** [comment]
Ah.. I see. Can I ask you some more detail about this?

First, I think we need to splilt data into three chunk, train, validation, test.

When we use determine the hyperparameter of model, only train dataset is used to train and evaluation will be done by using validation dataset.

My question is this.

When we evaluate the test dataset, what is the training dataset? only train dataset or union of train dataset and validation dataset?

I think both will be reasonable and the latter will get improved result.

**Comment by graytowne** [comment]
Should be the union of train and validation.
I think a good example is Kaggle, when you are on kaggle, you got a dataset, say D. And you usually split the data locally as: D -> (D_train, D_valid) to train model and do hyperparameter search. After you got the best configuration of hyperparameter, you have to use D as a whole to re-train a model and make predictions, otherwise you will loss performance as you didn't fully use the D.
Hope this example can answer your question.

**Comment by graytowne** [comment]
I am closing this issue now.
